### PR TITLE
Documentation improvements in publishing chapter

### PR DESCRIPTION
**Pull request by ljacomet**
Rework chapter structure and document `suppress*Warnings` for publication warnings.

### CODE REVIEW

File: subprojects/docs/src/docs/userguide/dep-man/06-publishing/publishing.adoc [hunk 1]
```diff
@@ -299,12 +295,44 @@ include::sample[dir="userguide/dependencyManagement/modelingFeatures/crossProjec
 include::sample[dir="userguide/dependencyManagement/modelingFeatures/crossProjectPublications/advanced-published/kotlin/buildSrc/src/main/kotlin/com/acme",files="InstrumentedJarsPlugin.kt[tags=add_variant_to_existing_component]"]
 ====
 
+[[sec:publishing_custom_artifacts_to_maven]]
+== Adding custom artifacts to a publication
+
+[WARNING]
```
Comment (jjohannes) [inline subprojects/docs/src/docs/userguide/dep-man/06-publishing/publishing.adoc]: We don't need to address this now, but I think this text is really wrong now. If anything, it should be turned around starting with Gradle metadata and then mentioning limitations of poms.

> remember that they are _secondary_ artifacts that support a _primary_ artifact.

I think the destination between primary artifact and other artifact should no longer be made (or only in a note about pom specifics).

> Thinking about publications in this way should help you determine whether you should be adding custom artifacts to an existing publication, or defining a new publication.

This is also much to simple in in my mind a misleading advice. Adding additional variants or feature variants is the correct model in most cases.

One simple thing we can do is at least not mark this as a warning...

Comment (ljacomet) [reply subprojects/docs/src/docs/userguide/dep-man/06-publishing/publishing.adoc]: I have reworked that WARNING block, let me know what you think!

File: subprojects/docs/src/docs/userguide/dep-man/06-publishing/publishing.adoc [hunk 2]
```diff
@@ -300,13 +300,16 @@ include::sample[dir="userguide/dependencyManagement/modelingFeatures/crossProjec
 
 [WARNING]
 ====
-When you're attaching extra artifacts to a publication, remember that they are _secondary_ artifacts that support a _primary_ artifact. The metadata that a publication defines — such as dependency information — is associated with that primary artifact only. Thinking about publications in this way should help you determine whether you should be adding custom artifacts to an existing publication, or defining a new publication.
-
-To some extent, you can see those extra artifacts as being published "out of context".
-The only way to address them as a consumer is to either to download them directly or use a _classifier_ notation.
+Instead of thinking in terms of artifacts, you should really embrace the variant aware model of Gradle.
```
Comment (jjohannes) [inline subprojects/docs/src/docs/userguide/dep-man/06-publishing/publishing.adoc]: 'really embrace' -> 'embrace'

File: subprojects/docs/src/docs/userguide/dep-man/06-publishing/publishing.adoc [hunk 3]
```diff
@@ -300,13 +300,16 @@ include::sample[dir="userguide/dependencyManagement/modelingFeatures/crossProjec
...
+It is expected that a single module may need multiple artifacts.
+However this rarely stops there, if the additional artifacts represent an <<feature_variants.adoc#,optional feature>>, they might also have different dependencies and more.
 
 Gradle, via _Gradle Module Metadata_, supports the publication of _additional variants_ which make those artifacts known to the dependency resolution engine.
 Please refer to the <<modeling_features.adoc#sec:variant-aware-sharing, variant-aware sharing>> section of the documentation to see how to declare such variants and <<#sec:publishing-custom-components, check out how to publish custom components>>.
+
+To some extent, you can see those extra artifacts as being published "out of context".
+If you attach extra artifacts to a publication, remember that they are _secondary_ artifacts only.
+The Maven metadata will only allow addressing these artifacts through the `classifier`.
```
Comment (jjohannes) [inline subprojects/docs/src/docs/userguide/dep-man/06-publishing/publishing.adoc]: Reorder sentences for clarification (?):

If you attach extra artifacts to a publication directly, they are published "out of context".
That means, they are not referenced in the metadata at all and can then only be addressed directly through a _classifier_ on a dependency.
Note that, in contrast to Gradle Module Metadata, Maven pom metadata will not contain information on additional artifacts regardless of whether they are added though a variant or directly, as variants can not be represented in the pom format.

**Comment by ljacomet** [comment]
Messed up command line merge so GitHub believes it was closed but I confirm this got merged!

**Pull request closed without merging**